healthbox3/healtbox3.py

```diff
@@ -0,0 +1,446 @@
+                room       =sensor['basic_id']
+            )
+            dataframe_list.append(dataframe)
+
+        # get sensor per room information
+        for key, roomnr in data['room'].items(): # loop over the available rooms
+            for sensor in roomnr['sensor']: # dive into sensors per room
+                # jump into parameter -> first dict in this dict -> get unit and value
+                for key, value in sensor['parameter'].items():
+                    sub_key   = key
+                    sub_value = value['value']
+                    sub_unit  = value['unit']
+                    identifier = str(sensor['basic_id']) + ' - ' +  str(sensor['name'] + ' - ' + str(sub_key))
+
+                    dataframe = DataFrame(
+                        identifier =identifier, 
+                        name       =sensor['type'], 
+                        value      =sub_value, 
+                        unit       =sub_unit, 
+                        room       =sensor['basic_id']
+                    )
+                    dataframe_list.append(dataframe)
+        return dataframe_list
+
+    def _sync_variables(self):
+        # Function to sync variables from HealthBox3 to gateway cache
+        # Get values from healthbox
+        resp = self._perform_get_request(HealthBox3Driver.Endpoints.DATA)
+        if resp is None:
+            return
+        if resp.status_code != 200:
+            return
+        data_json = resp.json()
+
+        # process values
+        dataframe_list = self._extract_data(data_json)
+        # upsert values
+        self.hbs.upsert_from_dataframe_list(dataframe_list)
+
+    def _sync_periodically(self):
+        """ General sync function to sync all data with the device """
+        self.logger("Starting to sync ")
+        while self.is_running:
+            try:
+                time.sleep(self.sync_delay)
+                self._sync_variables()
+            except KeyboardInterrupt:
+                self.stop()
+
+    class Endpoints:
+        def __init__(self):
+            pass
+        DATA = '/v2/api/data/current'
+        
+class HealthBox3Manager:
+    LISTEN_PORT = 49152  # The port to listen on to receive the reply packages back
+    DISCOVER_PORT = 49152  # The port to send the discover package to
+    DISCOVER_IP = '255.255.255.255'
+    DISCOVER_MESSAGE = 'RENSON_DEVICE/JSON?'
+    DEVICE_TYPE = 'HEALTHBOX3'
+
+    def __init__(self, logger):
+        # discovery elements
+        self.is_discovering = False
+        self.discovered_devices = [] # List of ip's
+        self.discover_socket = None
+        self.discover_callback = None
+
+        self.logger = logger
+
+        self.send_discovery_packet_thread = Thread(target=self._send_discovery_packet)
+        self.send_discovery_packet_thread.daemon = True
+
+    def _send_discovery_packet(self):
+        while self.is_discovering:
+            self.discover_socket.sendto(
+                b'RENSON_DEVICE/JSON?',
```

Comment:
It is very nice to break out the variables into class variables! I think you forgot to change it here to that variable name.